numba/cuda/dispatcher.py

```diff
@@ -358,6 +358,18 @@ def _prepare_args(self, ty, val, stream, retr, kernelargs):
             kernelargs.append(ctypes.c_double(val.real))
             kernelargs.append(ctypes.c_double(val.imag))
 
+        elif isinstance(ty, types.EnumMember):
+            try:
+                self._prepare_args(
+                    ty.dtype, val.value, stream, retr, kernelargs
+                )
+            except Exception as e:
+                if isinstance(e, NotImplementedError):
+                    raise NotImplementedError(
+                        f"Unsupported value for Enum {val}"
+                    ) from e
+                raise
```

Comment:
Could this be simplified to:

```suggestion
            except NotImplementedError as e:
                raise NotImplementedError(f"Unsupported value for Enum {val}")
```

(or is there a different effect I've missed here?)

Reply:
I was basically unfamiliar with the type of exception may be raised here. If only `NotImplementedError` is thrown then we can simplify this.